### PR TITLE
[DFT] Tidy examples code

### PR DESCRIPTION
**Pull request by hjabird**
* Remove errors from example code.
* Changes examples to use buffers and USM.
* Changes examples to use real and complex domain DFT.
* Changes compile-time dispatch example to use MKLGPU backend (currently most complete DFT backend).

### CODE REVIEW

**Review by fabiomestre — APPROVED** [review]

File: examples/dft/compile_time_dispatching/complex_fwd_buffer_mklgpu.cpp [hunk 1]
```diff
@@ -43,80 +28,76 @@
...
 #include "oneapi/mkl.hpp"
 
-// local includes
-#include "example_helper.hpp"
-
-void run_getrs_example(const sycl::device& cpu_device) {
-    // Matrix sizes and leading dimensions
-    constexpr std::size_t N = 10;
-    std::int64_t rs[3] {0, N, 1};
+void run_example(const sycl::device& gpu_deviceice) {
```
Comment (Rbiessy) [inline examples/dft/compile_time_dispatching/complex_fwd_buffer_mklgpu.cpp]: `gpu_deviceice` -> `gpu_device`

File: examples/dft/compile_time_dispatching/complex_fwd_buffer_mklgpu.cpp [hunk 2]
```diff
@@ -43,80 +28,76 @@
...
-    // oneapi::mkl::dft::compute_forward(desc, x_usm);
+    desc.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                   oneapi::mkl::dft::config_value::NOT_INPLACE);
+    desc.set_value(oneapi::mkl::dft::config_param::NUMBER_OF_TRANSFORMS,
+                   static_cast<std::int64_t>(1));
+
+    // 3. commit_descriptor (compile_time MKLGPU)
+    desc.commit(oneapi::mkl::backend_selector<oneapi::mkl::backend::mklgpu>{ gpu_queue });
+
+    // 5. compute_forward / compute_backward (MKLGPU)
```
Comment (Rbiessy) [inline examples/dft/compile_time_dispatching/complex_fwd_buffer_mklgpu.cpp]:
```suggestion
    // 4. compute_forward / compute_backward (MKLGPU)
```
(or just remove the numbering)

File: examples/dft/run_time_dispatching/real_fwd_usm.cpp [hunk 3]
```diff
@@ -42,33 +47,34 @@ void run_uniform_example(const sycl::device& dev) {
...
-    desc_vector.set_value(oneapi::mkl::dft::config_param::NUMBER_OF_TRANSFORMS, 4);
-    desc_vector.set_value(oneapi::mkl::dft::config_param::BWD_DISTANCE, N);
-    desc_vector.set_value(oneapi::mkl::dft::config_param::PLACEMENT, oneapi::mkl::dft::config_value::NOT_INPLACE);
+    desc.set_value(oneapi::mkl::dft::config_param::FORWARD_SCALE, 1.f / N);
+    desc.set_value(oneapi::mkl::dft::config_param::NUMBER_OF_TRANSFORMS,
+                   static_cast<std::int64_t>(1));
+    desc.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                   oneapi::mkl::dft::config_value::INPLACE);
+
+    // 4. commit_descriptor (runtime xPU)
```
Comment (Rbiessy) [inline examples/dft/run_time_dispatching/real_fwd_usm.cpp]: There's no step 3 there. `xPU`?

Comment (hjabird) [reply examples/dft/run_time_dispatching/real_fwd_usm.cpp]: Renumbered steps in [95f4e2c](https://github.com/hjabird/oneMKL/pull/6/commits/95f4e2ce326ac7a1c78068056ff9bbf3a400d705). I've replaced *runtime xPU* with *runtime dispatch*.

File: examples/dft/compile_time_dispatching/complex_fwd_buffer_mklgpu.cpp (hunk 2 above)
Comment (hjabird) [reply examples/dft/compile_time_dispatching/complex_fwd_buffer_mklgpu.cpp]: Done in [95f4e2c](https://github.com/hjabird/oneMKL/pull/6/commits/95f4e2ce326ac7a1c78068056ff9bbf3a400d705).

File: examples/dft/compile_time_dispatching/complex_fwd_buffer_mklgpu.cpp (hunk 1 above)
Comment (hjabird) [reply examples/dft/compile_time_dispatching/complex_fwd_buffer_mklgpu.cpp]: Done in [95f4e2c](https://github.com/hjabird/oneMKL/pull/6/commits/95f4e2ce326ac7a1c78068056ff9bbf3a400d705).

**Review by Rbiessy — APPROVED** [review]

**Comment by hjabird** [comment]
This PR is being merged into Codeplay/OneMKL in https://github.com/codeplaysoftware/oneMKL/pull/6.

**Pull request closed without merging**